ragna/source_storages/_lancedb.py

```diff
@@ -32,84 +39,81 @@ def __init__(self) -> None:
                     self._VECTOR_COLUMN_NAME,
-                    pa.list_(pa.float32(), self._embedding_dimensions),
+                    pa.list_(pa.float32(), len(documents[0].embedding)),
                 ),
                 pa.field("num_tokens", pa.int32()),
             ]
         )
 
-    _VECTOR_COLUMN_NAME = "embedded_text"
-
-    def store(
-        self,
-        documents: list[Document],
-        *,
-        chat_id: uuid.UUID,
-        chunk_size: int = 500,
-        chunk_overlap: int = 250,
-    ) -> None:
-        table = self._db.create_table(name=str(chat_id), schema=self._schema)
-
-        for document in documents:
-            for chunk in self._chunk_pages(
-                document.extract_pages(),
-                chunk_size=chunk_size,
-                chunk_overlap=chunk_overlap,
-            ):
-                table.add(
-                    [
-                        {
-                            "id": str(uuid.uuid4()),
-                            "document_id": str(document.id),
-                            "page_numbers": self._page_numbers_to_str(
-                                chunk.page_numbers
-                            ),
-                            "text": chunk.text,
-                            self._VECTOR_COLUMN_NAME: self._embedding_function(
-                                [chunk.text]
-                            )[0],
-                            "num_tokens": chunk.num_tokens,
-                        }
-                    ]
-                )
+        table = self._db.create_table(name=str(chat_id), schema=_schema)
+
+        for embedding in documents:
+            self._tokens += embedding.chunk.num_tokens
+            self._embeddings += 1
+            table.add(
+                [
+                    {
+                        "id": str(uuid.uuid4()),
+                        "document_id": str(embedding.chunk.document_id),
+                        "page_numbers": page_numbers_to_str(
+                            embedding.chunk.page_numbers
+                        ),
+                        "text": embedding.chunk.text,
+                        self._VECTOR_COLUMN_NAME: embedding.embedding,
+                        "num_tokens": embedding.chunk.num_tokens,
+                    }
+                ]
+            )
 
     def retrieve(
         self,
         documents: list[Document],
-        prompt: str,
+        prompt: list[float],
         *,
         chat_id: uuid.UUID,
-        chunk_size: int = 500,
         num_tokens: int = 1024,
     ) -> list[Source]:
         table = self._db.open_table(str(chat_id))
 
         # We cannot retrieve source by a maximum number of tokens. Thus, we estimate how
         # many sources we have to query. We overestimate by a factor of two to avoid
         # retrieving to few sources and needed to query again.
-        limit = int(num_tokens * 2 / chunk_size)
+        limit = int(num_tokens * 2 / self._tokens * self._embeddings)
```

Comment:
IIUC, we only have these instance variables to compute the approximate number of sources that we want to pull out, right? If so we can probably remove this and put a hardcoded limit here:

1. This is a micro-optimization. Later we are taking the right number anyway so this just limits the number of rows we _initially_ query. Without evidence that this actually improves performance significantly, I don't think we should complicate our code for that.
2. The estimation does not really work for Chroma and we hardcode a limit already

    https://github.com/Quansight/ragna/blob/a5e4795532b4ad96256a695fbd45ff83df8356be/ragna/source_storages/_chroma.py#L88-L100
3. Most importantly, after the chunking rework, I'm not even sure we can support this. If the chunking is not based on tokens, and that likely will be true for all chunking models except the one we have right now, we might not even have the chunking information available.

Meaning, I'm ok with hardcoding `100` here same as for Chroma and revisit on the chunking work.